pkg/libovsdb/schema.go

```diff
@@ -541,34 +545,110 @@ func (schemas *Schemas) LookupTable(dbname, table string) *TableSchema {
 
 /* validate */
 func (baseType *BaseType) ValidateInteger(value interface{}) bool {
-	_, ok := value.(int)
-	return ok
+	typeval, ok := value.(int)
+	if !ok {
+		return false
+	}
+	if baseType.Enum == nil {
+		return true
+	}
+	for _, v := range baseType.Enum {
+		enumval, ok := v.(int)
+		if !ok {
+			return false
+		}
+		if typeval == enumval {
+			return true
+		}
+	}
+	return false
 }
 
 func (baseType *BaseType) ValidateReal(value interface{}) bool {
-	_, ok := value.(float64)
-	return ok
+	typeval, ok := value.(float64)
+	if !ok {
+		return false
+	}
+	if baseType.Enum == nil {
+		return true
+	}
+	for _, v := range baseType.Enum {
+		enumval, ok := v.(float64)
+		if !ok {
+			return false
+		}
+		if typeval == enumval {
+			return true
+		}
+	}
+	return false
 }
 
 func (baseType *BaseType) ValidateBoolean(value interface{}) bool {
-	_, ok := value.(bool)
-	return ok
+	typeval, ok := value.(bool)
+	if !ok {
+		return false
+	}
+	if baseType.Enum == nil {
+		return true
+	}
+	for _, v := range baseType.Enum {
```

Comment:
I don't think that boolean can contain Enum, but it OK to have it just in case

Reply:
removed